x/concentrated-liquidity/incentives_test.go

```diff
@@ -3509,3 +3511,241 @@ func (s *KeeperTestSuite) TestPrepareBalancerPoolAsFullRange() {
+		},
+		"Balancer shares not yet added to CL pool accums": {
+			// 100 existing shares and 100 shares added from balancer
+			existingClLiquidity: defaultConcentratedAssets,
+			balancerPoolAssets:  defaultBalancerAssets,
+			uptimeGrowth:        uptimeHelper.hundredTokensMultiDenom,
+
+			balSharesNotAddedToAccums: true,
+			expectedError:             types.BalancerRecordNotFoundError{ClPoolId: 1, BalancerPoolId: 2, UptimeIndex: uint64(0)},
+		},
+	}
+	for name, tc := range tests {
+		s.Run(name, func() {
+			// --- Setup ---
+
+			// Set up CL pool with appropriate liquidity
+			s.SetupTest()
+			clPool := s.PrepareCustomConcentratedPool(s.TestAccs[0], tc.existingClLiquidity[0].Denom, tc.existingClLiquidity[1].Denom, DefaultTickSpacing, DefaultExponentAtPriceOne, sdk.ZeroDec())
+			clPoolId := clPool.GetId()
+
+			// Set up an existing full range position.
+			// Note that the second return value here is the position ID, not an error.
+			initialLiquidity, _ := s.SetupPosition(clPoolId, s.TestAccs[0], tc.existingClLiquidity[0], tc.existingClLiquidity[1], DefaultMinTick, DefaultMaxTick, s.Ctx.BlockTime())
+
+			// Create balancer pool to be linked with CL pool in happy path cases
+			balancerPoolId := s.PrepareCustomBalancerPool(tc.balancerPoolAssets, defaultBalancerPoolParams)
+
+			// Invalidate pool IDs if needed for error cases
+			if tc.balancerPoolDoesNotExist {
+				balancerPoolId = invalidPoolId
+			}
+			if tc.concentratedPoolDoesNotExist {
+				clPoolId = invalidPoolId + 1
+			}
+
+			// Link the balancer and CL pools
+			s.App.GAMMKeeper.SetMigrationInfo(s.Ctx,
+				gammtypes.MigrationRecords{
+					BalancerToConcentratedPoolLinks: []gammtypes.BalancerToConcentratedPoolLink{
+						{BalancerPoolId: balancerPoolId, ClPoolId: clPoolId},
+					},
+				})
+
+			// Add balancer shares to CL accumulatores
+			addedLiquidity := sdk.ZeroDec()
+			if !tc.balSharesNotAddedToAccums {
+				addedBalPool, qualifiedShares, err := s.App.ConcentratedLiquidityKeeper.PrepareBalancerPoolAsFullRange(s.Ctx, clPool.GetId())
+				addedLiquidity = addedLiquidity.Add(qualifiedShares)
+
+				// If a valid link exists, ensure no error and sanity check the output pool ID
+				if !tc.concentratedPoolDoesNotExist && !tc.balancerPoolDoesNotExist {
+					s.Require().NoError(err)
+					s.Require().Equal(addedBalPool, balancerPoolId)
+				}
+			}
+
+			// Emit incentives to the uptime accumulators
+			for _, growth := range tc.uptimeGrowth {
+				decEmissions := growth.MulDec(initialLiquidity.Add(addedLiquidity))
+				normalizedEmissions := sdk.NormalizeCoins(decEmissions)
+				s.FundAcc(clPool.GetIncentivesAddress(), normalizedEmissions)
+			}
+			addToUptimeAccums(s.Ctx, clPool.GetId(), s.App.ConcentratedLiquidityKeeper, tc.uptimeGrowth)
+
+			// --- System under test ---
+
+			amountClaimed, err := s.App.ConcentratedLiquidityKeeper.ClaimAndResetFullRangeBalancerPool(s.Ctx, clPoolId, balancerPoolId)
```

Comment:
I'm not seeing an assertion that `amountClaimed` is equal to expected when there is no error. IMO, this is blocking for this PR.

Reply:
Right – we currently check that `expectedAmount = amtAddedToGauge` when we should be checking `expectedAmount = amtAddedToGauge = amountClaimed`, nice catch